wallet/src/Pos/Wallet/Web/Methods/Logic.hs

```diff
@@ -157,37 +160,53 @@ getWallets = getWalletAddresses >>= mapM getWallet
 
 newAddress
     :: MonadWalletWebMode m
-    => AddrGenSeed
+    => IsBootstrapEraAddr
+    -> AddrGenSeed
     -> PassPhrase
     -> AccountId
     -> m CAddress
-newAddress addGenSeed passphrase accId =
+newAddress ibe addGenSeed passphrase accId =
     fixCachedAccModifierFor accId $ \accMod -> do
         -- check whether account exists
         _ <- getAccount accMod accId
 
-        cAccAddr <- genUniqueAccountAddress addGenSeed passphrase accId
+        cAccAddr <- genUniqueAccountAddress ibe addGenSeed passphrase accId
         addWAddress cAccAddr
         getWAddress accMod cAccAddr
 
+newAddressHandler
+    :: MonadWalletWebMode m
+    => AddrGenSeed
+    -> PassPhrase
+    -> AccountId
+    -> m CAddress
+-- TODO: AJ: Hard coding isBootstrapEra as True for API calls. Check if that's okay.
+newAddressHandler = newAddress (IsBootstrapEraAddr True)
+
 newAccountIncludeUnready
     :: MonadWalletWebMode m
-    => Bool -> AddrGenSeed -> PassPhrase -> CAccountInit -> m CAccount
-newAccountIncludeUnready includeUnready addGenSeed passphrase CAccountInit {..} =
+    => Bool -> IsBootstrapEraAddr -> AddrGenSeed -> PassPhrase -> CAccountInit -> m CAccount
+newAccountIncludeUnready includeUnready ibe addGenSeed passphrase CAccountInit {..} =
     fixCachedAccModifierFor caInitWId $ \accMod -> do
         -- check wallet exists
         _ <- getWalletIncludeUnready includeUnready caInitWId
 
         cAddr <- genUniqueAccountId addGenSeed caInitWId
         createAccount cAddr caInitMeta
-        () <$ newAddress addGenSeed passphrase cAddr
+        () <$ newAddress ibe addGenSeed passphrase cAddr
         getAccount accMod cAddr
 
 newAccount
     :: MonadWalletWebMode m
-    => AddrGenSeed -> PassPhrase -> CAccountInit -> m CAccount
+    => IsBootstrapEraAddr -> AddrGenSeed -> PassPhrase -> CAccountInit -> m CAccount
 newAccount = newAccountIncludeUnready False
 
+newAccountHandler
+    :: MonadWalletWebMode m
+    => AddrGenSeed -> PassPhrase -> CAccountInit -> m CAccount
+-- TODO: AJ: Hard coding isBootstrapEra as True for API calls. Check if that's okay.
```

Comment:
this looks ok for me. @flyingleafe right?

Reply:
I kept it that way because I didn't want to change the existing behaviour of the API. But do you think it makes sense to add another address endpoint which allows controlling the value for isBootstrapEra?

Reply:
No, I think that here `getBootstrapEra` should be used too. This way we won't change the behavior for bootstrap era and we'll have correct behavior when reward era comes.